src/AutoRest.CSharp/Common/Generation/Writers/RestClientWriter.cs

```diff
@@ -17,6 +18,9 @@ namespace AutoRest.CSharp.Generation.Writers
 {
     internal class RestClientWriter
     {
+        internal delegate void WritePageFuncBody(CodeWriter writer, CodeWriterDeclaration messageVariable, RestClientMethod operation, string pipelineName, bool async);
+        internal delegate void WriteStatusCodeImplementation(CodeWriter writer, string messageVariable, RestClientMethod operation, bool async, FieldDeclaration fieldDeclaration);
```

Comment:
Also I think if it's overridden in sub-classes, you should consider make those as abstract methods, let sub-classes implement them and remove `static`. We have too many static methods in writers.

Reply:
@archerzz which static method you mean?

Reply:
@pshao25 The implementation of the delegate. I think since they are implemented by sub-classes, those two should be declared as virtual. delegate is more for external interaction.

Reply:
The problem is: `LowLevelClientWriter` is not inheriting `RestClientWriter`. I'll move `WriteFuncBodyWithProcessMessage` to `LowLevelClientWriter`.